examples/website/i3s-building-scene-layer/src/components/building-explorer.jsx

```diff
@@ -0,0 +1,166 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import {Font, Color} from './styles';
+import {Checkbox, CheckboxOption, CheckboxSpan} from './checkbox';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faAngleRight, faAngleDown, faCircle} from '@fortawesome/free-solid-svg-icons';
+import ToggleSwitch from './toggle-switch';
+
+const BuildingExplorerContainer = styled.div`
+  height: ${(props) => (props.isShown ? '450px' : '40px')};
+  overflow: auto;
+  align-items: flex-start;
+  position: absolute;
+  top: 140px;
+  display: flex;
+  flex-direction: column;
+  background: #0e111a;
+  border-radius: 8px;
+  padding-top: 4px;
+  @media (max-width: 768px) {
+    top: 63px;
+    width: 100%;
+    border-radius: 0;
+  }
+`;
+
+const BuildingExplorerSublayers = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const CollapseContainer = styled.div`
+  margin-right: 5px;
+  cursor: pointer;
+`;
+
+const CheckboxContainer = styled.div`
+  ${Color}
+  ${Font}
+  margin-left: 10px;
+`;
+
+const Label = styled.h3`
+  margin: 0 16px 8px 16px;
+  cursor: pointer;
+  color: white;
+  font-weight: normal;
+`;
+
+const propTypes = {
+  sublayers: PropTypes.array,
+  onToggleBuildingExplorer: PropTypes.func,
+  isShown: PropTypes.bool
+};
+const defaultProps = {
+  sublayers: [],
+  isShown: false
+};
+
+export default class BuildingExplorer extends PureComponent {
```

Comment:
I suspect that it is time to start building functional React components with hooks? This is very old style React, for a new example it doesn't really make sense?

It could be a follow-up PR. I already converted some examples like geospatial and tiles to hooks so there is something to compare with.

Reply:
We've raised a story for porting to React Functional Components